### ISSUE TITLE
Outdated Sinatra fails to boot

### ISSUE THREAD

**Issue by dekubu** [rack-app/rack-app#55]
Given 

```
require 'rack/app'
require_relative 'sinatra'
require_relative "./dlr/config/environment" # small rails app


class App < Rack::App

  mount_rack_based_application Dlr::Application, to: "/rails" 
  mount_rack_based_application SinatraApp, to: "/sinatra"

  Rails.application.load_server
  
  desc 'health check endpoint'
  get '/' do
    'OK'
  end

  get '/hello' do
    'Hello World'
  end

end
```
[when]([url](url)) I run 

```
bundle exec rackup config.ru
```

I get the following Error

![Screenshot 2023-10-04 at 09 21 07](https://github.com/rack-app/rack-app/assets/141216207/16830dae-4636-425a-9540-1d74c9e0a902)

on investigation in noticed that when I bundle it is rolling back to version 7 of rack-app, admiteldly I am not sure how relevant it is.

![Screenshot 2023-10-04 at 09 12 37](https://github.com/rack-app/rack-app/assets/141216207/9fd9b7c5-4b59-4880-b70a-cf6fef5c5bd5)

here is a zip of the project.

[broken-rack-app.zip](https://github.com/rack-app/rack-app/files/12800094/broken-rack-app.zip)

Any help would be hugely appreciated.

**Comment by adamluzsi** [comment]
Hi @dekubu,

I'd be glad to assist in any way possible.

It seems like the version of Sinatra you have is locked at v1 which is very outdated, as the current version is v3. The error you're encountering is a known issue with that older version. I'd recommend investigating what might have locked Sinatra to that version, as it doesn't appear to be related to rack-app. The app seems to be failing during boot due to the loading of this outdated Sinatra version.

Feel free to reach out if you need further assistance.

Warm regards,
Adam

**Comment by dekubu** [comment]
Hey, thanks again for the help l this has been such an amazing learning journey since I have picked up rack-app, I have definitely become a better developer and I am actually starting to understand rails, I can see what it does for me and what it doesn't. 

I find that it is better to know what my software is doing rather than just delagate it to the framework, I find When shit does wrong I can actually work things and by solving the problems I feel way more confident now about when I deploy ! so cant thank you enough for the help.

I have solved the problems now , I upgraded all my dependencies, and systematically fixed the errors one by one and now I have  a rack-app with a rails app mounted, this allows me the flexibility to use all the goodies of rails as I need it while the majority of my work is done in rack-app! 

I have been encapsulating my learning into this 

https://gitlab.com/dekubu/hud 

which I hope in time to do write up about! 

help is much appreciated ! and we can now close this one.

I am always asking lol so if I can help in anyway shout .
